File: tests/fixtures/agent_generated/agent_sql_format.ts
// Agent-generated: AI assistant uses template literals in SQL queries
// Expected: SEC-015 (sql_injection_ts)

import express from "express";

const app = express();
const db: any = {};  // placeholder

app.get("/api/users/search", async (req, res) => {
  const query = req.query.q as string;
  // AI generated: template literal SQL injection
  const sql = `SELECT * FROM users WHERE name LIKE '%${query}%'`;
  const results = await db.query(sql);
  res.json(results);
});

app.get("/api/users/:id", async (req, res) => {
  const id = req.params.id;
  // AI generated: string concatenation in SQL
  const result = await db.query("SELECT * FROM users WHERE id = " + id);
  res.json(result);
});

app.get("/api/orders", async (req, res) => {
  const sortBy = req.query.sort || "created_at";
  // AI generated: user input in ORDER BY clause
  const sql = `SELECT * FROM orders ORDER BY ${sortBy}`;
  const results = await db.query(sql);
  res.json(results);
});

File: tests/fixtures/agent_generated/agent_sql_format.py
# Agent-generated: AI assistant uses f-strings for SQL queries
# Expected: SEC-015 (sql_injection)

from flask import Flask, jsonify, request
import sqlite3

app = Flask(__name__)
DB_PATH = "app.db"

def get_db():
    return sqlite3.connect(DB_PATH)

@app.route("/api/users/search")
def search_users():
    query = request.args.get("q", "")
    db = get_db()
    # AI generated: direct f-string interpolation in SQL
    cursor = db.execute(f"SELECT * FROM users WHERE name LIKE '%{query}%'")
    results = cursor.fetchall()
    return jsonify(results)

@app.route("/api/users/<username>")
def get_user(username):
    db = get_db()
    # AI generated: format string in SQL
    cursor = db.execute("SELECT * FROM users WHERE username = '%s'" % username)
    return jsonify(cursor.fetchone())

@app.route("/api/orders")
def get_orders():
    sort_by = request.args.get("sort", "created_at")
    db = get_db()
    # AI generated: user-controlled ORDER BY
    cursor = db.execute(f"SELECT * FROM orders ORDER BY {sort_by}")
    return jsonify(cursor.fetchall())
